### PR TITLE
Add node-server

### PR DESCRIPTION
**Pull request by arikmaor**
This also includes updates to the `web-ui` image and README

After this I've successfully created an account and logged in to `web-ui`!

```
$ kubectl get all
NAME                                     READY   STATUS    RESTARTS   AGE
pod/elasticsearch-master-0               1/1     Running   0          87m
pod/kibana-7dd74f6d55-kzdnw              1/1     Running   0          87m
pod/tensorleap-web-ui-68f4bfc7dd-nc6x8   1/1     Running   0          55m
pod/mongodb-5bd768db8-4s9m2              1/1     Running   0          49m
pod/node-server-594d8d657f-4wjlq         1/1     Running   0          44m

NAME                                    TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)             AGE
service/kubernetes                      ClusterIP   10.152.183.1     <none>        443/TCP             97m
service/tensorleap-web-ui               NodePort    10.152.183.104   <none>        8080:30552/TCP      87m
service/kibana                          ClusterIP   10.152.183.29    <none>        5601/TCP            87m
service/elasticsearch-master            ClusterIP   10.152.183.88    <none>        9200/TCP,9300/TCP   87m
service/elasticsearch-master-headless   ClusterIP   None             <none>        9200/TCP,9300/TCP   87m
service/mongodb                         ClusterIP   10.152.183.222   <none>        27017/TCP           49m
service/node-server                     NodePort    10.152.183.169   <none>        80:30979/TCP        49m

NAME                                READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/kibana              1/1     1            1           87m
deployment.apps/tensorleap-web-ui   1/1     1            1           87m
deployment.apps/mongodb             1/1     1            1           49m
deployment.apps/node-server         1/1     1            1           49m

NAME                                           DESIRED   CURRENT   READY   AGE
replicaset.apps/kibana-7dd74f6d55              1         1         1       87m
replicaset.apps/tensorleap-web-ui-68f4bfc7dd   1         1         1       55m
replicaset.apps/tensorleap-web-ui-8c67768d8    0         0         0       87m
replicaset.apps/mongodb-5bd768db8              1         1         1       49m
replicaset.apps/node-server-594d8d657f         1         1         1       44m
replicaset.apps/node-server-544ffd5d87         0         0         0       49m

NAME                                    READY   AGE
statefulset.apps/elasticsearch-master   1/1     87m
```

### CODE REVIEW

**Review by yotamazriel — APPROVED** [review]
LGTM

File: constructs/node-server.ts [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+import {
+  IntOrString,
+  KubeDeployment,
+  KubeIngress,
+  KubeRole,
+  KubeService,
+  KubeServiceAccount,
+  PolicyRule,
+} from '../imports/k8s';
+import { KubeRoleBinding } from 'cdk8s-plus-22/lib/imports/k8s';
```
[the remark this answers is not in the document]
Comment (arikmaor) [reply constructs/node-server.ts:12]: It's a mistake, 10x
fixing

File: constructs/node-server.ts [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      },
+    });
+
+    new MongoDB(this, 'mongodb');
+
+    const { serviceAccount } = new KubeServiceAccountWithRole(
+      this,
+      'node-server-sa',
+      {
+        rules: [
```
[the remark this answers is not in the document]
Comment (arikmaor) [reply constructs/node-server.ts:32]: yes

File: constructs/node-server.ts [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+                  },
+                ],
+                env: [
+                  {
+                    name: 'NODE_ENV',
+                    value: 'production',
+                  },
+                  {
+                    name: 'KIBANA_URL',
+                    value: 'http://kibana.default:5601',
```
[the remark this answers is not in the document]
Comment (arikmaor) [reply constructs/node-server.ts:84]: I'll do a proper refactor after getting this working

File: constructs/node-server.ts [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+                  },
+                  {
+                    name: 'MONGO_URI',
+                    value:
+                      'mongodb://mongodb.default.svc/tensorleap?tls=false&ssl=false',
+                  },
+                  {
+                    name: 'NAMESPACE',
```
[the remark this answers is not in the document]
Comment (arikmaor) [reply constructs/node-server.ts:92]: it's a native `k8s` way of getting the `namespace` in an environment variable
What's the problem?

File: constructs/node-server.ts [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+}
+
+export class MongoDB extends Chart {
+  constructor(scope: Construct, id: string) {
+    super(scope, id, {
+      namespace: 'default',
+    });
+
+    new Helm(this, 'mongodb', {
+      chart: 'bitnami/mongodb',
```
[the remark this answers is not in the document]
Comment (arikmaor) [reply constructs/node-server.ts:255]: Its is already added, look at the changes to docker file in this PR

File: constructs/node-server.ts [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+                    value: 'production',
+                  },
+                  {
+                    name: 'KIBANA_URL',
+                    value: 'http://kibana.default:5601',
+                  },
+                  {
+                    name: 'MONGO_URI',
+                    value:
+                      'mongodb://mongodb.default.svc/tensorleap?tls=false&ssl=false',
```
[the remark this answers is not in the document]
Comment (arikmaor) [reply constructs/node-server.ts:89]: I'll do some refactoring later

**Pull request merged**